File: shared/src/main/java/ru/privatenull/pnauth/limbo/PicoLimboProvider.java
package ru.privatenull.pnauth.limbo;

public final class PicoLimboProvider implements LimboServerProvider {
    @Override
    public String id() {
        return "pico";
    }

    @Override
    public LimboServer create(LimboServerContext context) {
        return new PicoLimboServer(context.dataDirectory(), context.settings());
    }
}

File: shared/src/main/kotlin/ru/privatenull/pnauth/limbo/PicoLimboProvider.kt
package ru.privatenull.pnauth.limbo

import java.nio.file.Files

class PicoLimboProvider : LimboServerProvider {
    override fun id(): String = "pico"

    override fun create(context: LimboServerContext): LimboServer {
        val store = PicoLimboConfigStore()
        val limboFolder = context.dataDirectory.resolve("limbo")
        Files.createDirectories(limboFolder)
        val configFile = limboFolder.resolve("pico_limbo.toml")
        val config = if (Files.exists(configFile)) store.load(configFile) else PicoLimboConfig()
        return PicoLimboServer(limboFolder, limboFolder, context.settings, config)
    }
}
